Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+      header: "Created By",
+      id: "createdBy",
+      cell: (value) => (
+        <>
+          {value.row.original.createdBy && (
+            <>
+              {value.row.original.createdBy.firstName}{" "}
+              {value.row.original.createdBy.lastName} (
+              {value.row.original.createdBy.email})
+            </>
+          )}
+          {!value.row.original.createdBy && <>System</>}
+        </>
+      ),
+    }),
+  ];
+
+  // async function so the form can manage the loading state
+  const submitOrder = async (updatedOrder: OrderModel) => {
+    const editedOrder: OrderModel = {
+      // uneditable fields
+      id: updatedOrder.id,
+      status: updatedOrder.status,
+      createdOn: updatedOrder.createdOn,
+      total: updatedOrder.total,
+      subTotal: updatedOrder.subTotal,
+      balanceRemaining: updatedOrder.balanceRemaining,
+      billings: updatedOrder.billings,
+      payments: updatedOrder.payments,
+      history: updatedOrder.history,
+      piUser: updatedOrder.piUser,
+      percentTotal: updatedOrder.percentTotal,
+      nextPaymentDate: updatedOrder.nextPaymentDate,
+
+      // editable fields
+      PILookup: updatedOrder.PILookup,
+      name: updatedOrder.name,
+      productName: updatedOrder.productName,
+      description: updatedOrder.description,
+      category: updatedOrder.category,
+      externalReference: updatedOrder.externalReference,
+      notes: updatedOrder.notes,
+      units: updatedOrder.units,
+      unitPrice: updatedOrder.unitPrice,
+      quantity: updatedOrder.quantity,
+      installments: updatedOrder.installments,
+      installmentType: updatedOrder.installmentType,
+      adjustment: updatedOrder.adjustment,
+      adjustmentReason: updatedOrder.adjustmentReason,
+      adminNotes: updatedOrder.adminNotes,
+      metaData: updatedOrder.metaData,
+      lifeCycle: updatedOrder.lifeCycle,
+      expirationDate: updatedOrder.expirationDate,
+      installmentDate: updatedOrder.installmentDate,
+    };
+
+    // TODO: await API call
+    // const newOrder = await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    setOrder(editedOrder); // should be newOrder once it's pulling from the API
+  };
+
+  const [editPaymentModel, setEditPaymentModel] = useState<PaymentModel>({
+    id: 0,
+    amount: 0,
+    entryAmount: "",
+    status: "",
+    createdOn: "",
+  });
+
+  const [makePaymentConfirmation] = useConfirmationDialog<PaymentModel>(
+    {
+      title: "Make One Time Payment",
+      message: (setReturn) => (
+        <>
+          <div className="form-group">
+            <label htmlFor="fieldAmount">Amount</label>
+            <input
+              className="form-control"
+              id="fieldAmount"
+              required
+              value={editPaymentModel.entryAmount}
+              onChange={(e) => {
+                const value = e.target.value;
+                if (/^\d*\.?\d*$/.test(value) || /^\d*\.$/.test(value)) {
```

Comment:
good but you can probably also just use input type number w/ some parameters (step, min/max) to validate in real-time

Reply:
#395